### PR TITLE
Fix/seperate high and secure

### PR DESCRIPTION
**Pull request by ErebusZ**
This Pr is Failling due to missing kb entry, [KB entry PR](https://github.com/Ostorlab/KB/pull/130)

### CODE REVIEW

File: agent/process_scans.py [hunk 1]
```diff
@@ -37,3 +37,18 @@ def get_technical_details(scans: dict[str, Any], target: str | None) -> str:
         technical_detail = f"Analysis of the target `{target}`:\n"
     technical_detail += markdown.table_markdown(formatted_scans)
     return technical_detail
+
+
+def split_scans_by_result(
```
Comment (PiranhaSa) [inline agent/process_scans.py:42]: if it's a public function it will need a unit test

File: agent/process_scans.py [hunk 2]
```diff
@@ -37,3 +37,18 @@ def get_technical_details(scans: dict[str, Any], target: str | None) -> str:
...
+    scans: dict[str, Any],
+) -> tuple[dict[str, Any], dict[str, Any]]:
+    secure_scans: dict[str, Any] = dict()
```
Comment (deadly-panda) [inline agent/process_scans.py:45]: a micro-optimisation to keep in mind: `{}` is faster than `dict()`.

File: agent/virus_total_agent.py [hunk 3]
```diff
@@ -87,13 +87,31 @@ def _process_response(self, response: dict[str, Any], target: str | None) -> Non
...
-                    entry=kb.KB.VIRUSTOTAL_SCAN,
-                    technical_detail=technical_detail,
-                    risk_rating=risk_rating,
-                )
+                (
+                    secure_scan_report,
+                    vulnerable_scan_report,
+                ) = process_scans.split_scans_by_result(scans)
+
+                if len(secure_scan_report) != 0:
```
Comment (deadly-panda) [inline agent/virus_total_agent.py:95]:
```suggestion
                if len(secure_scan_report) > 0:
```

File: agent/virus_total_agent.py [hunk 4]
```diff
@@ -87,13 +87,31 @@ def _process_response(self, response: dict[str, Any], target: str | None) -> Non
...
+                    technical_detail = process_scans.get_technical_details(
+                        secure_scan_report, target
+                    )
+                    self.report_vulnerability(
+                        entry=kb.KB.SECURE_VIRUSTOTAL_SCAN,
+                        technical_detail=technical_detail,
+                        risk_rating=agent_report_vulnerability_mixin.RiskRating.SECURE,
+                    )
+
+                if len(vulnerable_scan_report) != 0:
```
Comment (deadly-panda) [inline agent/virus_total_agent.py:105]:
```suggestion
                if len(vulnerable_scan_report) > 0:
```

File: agent/virus_total_agent.py [hunk 5]
```diff
@@ -87,13 +87,31 @@ def _process_response(self, response: dict[str, Any], target: str | None) -> Non
...
+                        technical_detail=technical_detail,
+                        risk_rating=agent_report_vulnerability_mixin.RiskRating.SECURE,
+                    )
+
+                if len(vulnerable_scan_report) > 0:
+                    technical_detail = process_scans.get_technical_details(
+                        vulnerable_scan_report, target
+                    )
+                    self.report_vulnerability(
+                        entry=kb.KB.INSECURE_VIRUSTOTAL_SCAN,
```
Comment (BlueSquare1) [inline agent/virus_total_agent.py:110]: Would be better to add scan link in KB references rather than `www.virustotal.com`

**Review by 3asm — APPROVED** [review]

**Pull request merged**